VCB-222: fix memory of seven segment display

deflox opened the pull request:
Before this fix, the seven segment display had zeros filled at positions that were not manually overridden by the user. This fix now enables the seven segment display to behave exactly like the real CT board: Per default FF is written for every byte at every memory address of the seven segment display.

PhilSchni commented:
@deflox : There are some failing tests

m-dot-wigger commented on src/board/devices/output/seven-segment.ts:
```diff
@@ -63,15 +63,8 @@ export class SevenSegmentDevice extends Device {
-
-  constructor() {
-    super()
-    this.writeWord(this.startAddress, Word.fromUnsignedInteger(0xffffffff))
-    this.isOn = false
-    this.isBinaryMode = false
-  }
```
just call reset in constructor so it behaves like before

deflox commented:
I have fixed the tests. I have removed the `isOn` flag as well because I realized that is not really needed as the seven segment mode initially sets itself to FF and is initially not in binary mode the result will be that no segment is glowing. In segment control mode the lights are off if everything is set to FF.

schuelihueli commented:
with the  newest changes
 `AREA MyCode, CODE, READONLY
	LDR R3,=0
	LDR R1, =0x11
	LDR R2, =0x60000110
	STRB R1, [R2, R3]`
does not show "F" on Display 1-3

deflox commented:
> with the newest changes `AREA MyCode, CODE, READONLY LDR R3,=0 LDR R1, =0x11 LDR R2, =0x60000110 STRB R1, [R2, R3]` does not show "F" on Display 1-3

Yes but this is the intended behavior. If you write to address `0x60000110` you will be in segment control mode. In that case, at address `0x60000110` your byte `0x11` is stored but at address `0x60000111` until `0x60000113` still `0xFF` is stored. Since we are in segment control mode and it is active-low, the other three display are not glowing at all because every bit is set to 1.

deflox commented:
> > > with the newest changes `AREA MyCode, CODE, READONLY LDR R3,=0 LDR R1, =0x11 LDR R2, =0x60000110 STRB R1, [R2, R3]` does not show "F" on Display 1-3
> > 
> > 
> > Yes but this is the intended behavior. If you write to address `0x60000110` you will be in segment control mode. In that case, at address `0x60000110` your byte `0x11` is stored but at address `0x60000111` until `0x60000113` still `0xFF` is stored. Since we are in segment control mode and it is active-low, the other three displays are not glowing at all because every bit is set to 1.

To my knowledge it behaves exactly the same on the CT board. Only if the display is set to binary control mode FF is shown on the other two displays if only one byte is written at address 114.
